pkg/controller/storageversionmigrator/storageversionmigrator.go

```
@@ -0,0 +1,289 @@
+	}) {
+		return fmt.Errorf("dependency graph is not synced, requeuing to attempt again")
+	}
+
+	gcListResourceVersion, err := convertResourceVersionToInt(svmc.gcCollector.GetMonitorController(gvr).LastSyncResourceVersion())
+	if err != nil {
+		return err
+	}
+	listResourceVersion, err := convertResourceVersionToInt(toBeProcessedSVM.Status.ResourceVersion)
+	if err != nil {
+		return err
+	}
+
+	if gcListResourceVersion < listResourceVersion {
+		return fmt.Errorf("GC cache is not up to date, requeuing to attempt again. gcListResourceVersion: %d, listResourceVersion: %d", gcListResourceVersion, listResourceVersion)
+	}
+
+	toBeProcessedSVM, err = updateSVMStatus(
+		ctx,
+		svmc.kubeClient,
+		toBeProcessedSVM,
+		svmv1alpha1.MigrationRunning,
+		migrationRunningStatusReason,
+	)
+	if err != nil {
+		return err
+	}
+
+	// ToDo: implement a mechanism to resume migration from the last migrated resource in case of a failure
+	// process storage migration
+	for _, gvrKey := range svmc.gcCollector.GetMonitorStore(gvr).ListKeys() {
+		namespace, name, err := cache.SplitMetaNamespaceKey(gvrKey)
+		if err != nil {
+			return err
+		}
+
+		typeMeta := metav1.TypeMeta{}
+		gvk, err := svmc.restMapper.KindFor(gvr)
+		if err != nil {
+			return err
+		}
+		typeMeta.APIVersion, typeMeta.Kind = gvk.ToAPIVersionAndKind()
+		data, err := json.Marshal(typeMeta)
+		if err != nil {
+			return err
+		}
+
+		_, err = svmc.dynamicClient.Resource(gvr).
+			Namespace(namespace).
+			Patch(ctx,
+				name,
+				types.ApplyPatchType,
+				data,
+				metav1.PatchOptions{
+					FieldManager: svmc.controllerName,
+				},
+			)
+		if err != nil {
+			// in case of NotFound or Conflict, we can stop processing migration for that resource
+			if errors.IsNotFound(err) || errors.IsConflict(err) {
+				continue
+			}
+
+			_, err = updateSVMStatus(
+				ctx,
+				svmc.kubeClient,
+				toBeProcessedSVM,
+				svmv1alpha1.MigrationFailed,
+				migrationFailedStatusReason,
```

why not report the error?  Concerns about leaking PII?  That's valid, but a category of error (http status code) is safe enough to report.